tldw_Server_API/app/api/v1/endpoints/chat.py

```
@@ -2084,3 +2061,90 @@ def _sanitize_json_for_rate_limit(request_json: str) -> str:
+)
+async def save_chat_knowledge(
+    payload: KnowledgeSaveRequest,
+    db: CharactersRAGDB = Depends(get_chacha_db_for_user),
+    current_user: User = Depends(get_request_user),
+):
+    """Persist a snippet from a conversation into Notes (and optional Flashcard)."""
+    try:
+        # Validate conversation ownership and optional message linkage before mutating.
+        conversation = db.get_conversation_by_id(payload.conversation_id)
+        if not conversation or conversation.get("deleted"):
+            raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Conversation not found")
```

Ruff (0.14.6)

Abstract `raise` to an inner function

(TRY301)